exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,679 @@
+	sb.WriteString(" ")
+	sb.WriteString(data.Name)
+
+	// Prefix the name with the type of RPC
+	data.Name = sb.String()
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Url = data.Name
+
+	sb.Reset()
+
+	writeFormattedPeerAddressFromNetworkAttributes(&attrs.NetworkAttributes, &sb)
+
+	data.Source = sb.String()
+}
+
+// Maps RPC Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md
+func fillRemoteDependencyDataRPC(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractRPCAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Data = data.Name
+
+	data.Type = attrs.RPCSystem
+
+	var sb strings.Builder
+	writeFormattedPeerAddressFromNetworkAttributes(&attrs.NetworkAttributes, &sb)
+	data.Target = sb.String()
+}
+
+// Maps Database Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md
+func fillRemoteDependencyDataDatabase(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractDatabaseAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	data.Type = attrs.DBSystem
+
+	if attrs.DBStatement != "" {
+		data.Data = attrs.DBStatement
+	} else if attrs.DBOperation != "" {
+		data.Data = attrs.DBOperation
+	}
+
+	var sb strings.Builder
+	writeFormattedPeerAddressFromNetworkAttributes(&attrs.NetworkAttributes, &sb)
+	data.Target = sb.String()
+}
+
+// Maps Messaging Consumer/Server Span to AppInsights RequestData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+func fillRequestDataMessaging(span pdata.Span, data *contracts.RequestData) {
+	attrs := copyAndExtractMessagingAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	// TODO Understand how to map attributes to RequestData fields
+	if attrs.MessagingURL != "" {
+		data.Source = attrs.MessagingURL
+	} else {
+		var sb strings.Builder
+		writeFormattedPeerAddressFromNetworkAttributes(&attrs.NetworkAttributes, &sb)
+		data.Source = sb.String()
+	}
+}
+
+// Maps Messaging Producer/Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+func fillRemoteDependencyDataMessaging(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractMessagingAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	// TODO Understand how to map attributes to RemoteDependencyData fields
+	data.Data = attrs.MessagingURL
+	data.Type = attrs.MessagingSystem
+
+	if attrs.MessagingURL != "" {
+		data.Target = attrs.MessagingURL
```

Comment:
message.destination is `Required` field on the spec, so referring to `messaging.destination` also makes sense to me. (I saw your previous pr referred to this attr.) how do you think?

1. messaging.url
2. messaging.destination
3. net.peer.*

![image](https://user-images.githubusercontent.com/11863108/86535171-32cd3280-be93-11ea-9e27-49b117490c54.png)

Reply:
makes sense. I did have it like that previously but seemed a little redundant because it also the Span name probably.

I’ll fix the tests also

Reply:
I just did some validations on appinsight. 

Let's assume that single publisher and multiple subscribers for topicA are tracked by the same appinsight and show the links between single source to multiple targets.

But publisher linked to one of them because Target might be designed for referring to single destination in appmap. 

So if users want to see links for single publisher and multiple subscribers on appmap, they should not set `messaging.url` and `net.peer.*` to keep Target empty.

If users want to see the link to dependency pubsub system regardless of topic, they can set `messaging.url` pr `net.peer.*`

So, the current change (referring to messaging.url and net.peer.*) is better fit for appinsight appmap. Apologies for the wrong suggestion. 

Once you fixed the unit-test, I will approve it. Thanks a lot for this work!

Reply:
Here is the screenshot. react-form publishes message to node-subscriber and python-subscriber via topic A and B. appmap shows the links between publisher and subscribers.

![image](https://user-images.githubusercontent.com/11863108/86538429-e130a200-beaa-11ea-8aa4-18929f121b92.png)

We can see the list of topic which node-subscriber subscribes.
![image](https://user-images.githubusercontent.com/11863108/86538450-fb6a8000-beaa-11ea-8833-2b9049c13d0e.png)

So the current change which doesn't refer to `messaging.destination` is better fit for app insight app map. If we use messaging.destination for Target, then we can see one link between react-form and node-subscriber or python-subscriber.